### PR TITLE
prosemd-lsp: init at 0.1.0

### PR DESCRIPTION
**Pull request by caarlos0**
###### Description of changes

adds prosemd-lsp.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ocfox — CHANGES_REQUESTED** [review]
Where is the top-level entry?

**Comment by caarlos0** [comment]
@ocfox done

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23074,6 +23074,8 @@ with pkgs;
     buildPythonApplication click future six;
   };
 
+  proselint-lsp = callPackage ../development/tools/misc/proselint-lsp {};
```
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:23077]: proselint?

Comment (caarlos0) [reply pkgs/top-level/all-packages.nix:23077]: that is what happens when you do 5 things at the same time, fixed, sorry about it, and thanks for pointing it out 🙏

File: pkgs/development/tools/misc/prosemd-lsp/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description =
+      "An experimental proofreading and linting language server for markdown files";
+    homepage = "https://github.com/kitten/prosemd-lsp";
+    license = licenses.lgpl21;
+    maintainers = [ maintainers.caarlos0 ];
+  };
+
+  passthru.updateScript = nix-update-script { };
```
Comment (ocfox) [inline pkgs/development/tools/misc/prosemd-lsp/default.nix:23]: It's better to move this above the meta.

Comment (caarlos0) [reply pkgs/development/tools/misc/prosemd-lsp/default.nix]: fixed

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 223488` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>prosemd-lsp</li>
  </ul>
</details>

File: pkgs/development/tools/misc/prosemd-lsp/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  cargoSha256 = "1KE84o/KDkDMwlxqeDQmniBL6mDE1RSWG5J8nTvBwa4=";
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    description =
+      "An experimental proofreading and linting language server for markdown files";
+    homepage = "https://github.com/kitten/prosemd-lsp";
+    license = licenses.lgpl21;
+    maintainers = [ maintainers.caarlos0 ];
```
Comment (ocfox) [inline pkgs/development/tools/misc/prosemd-lsp/default.nix:22]: This package should work on other platforms, may add them.

**Review by ocfox — APPROVED** [review]

File: pkgs/development/tools/misc/prosemd-lsp/default.nix (hunk 3 above)
Comment (caarlos0) [reply pkgs/development/tools/misc/prosemd-lsp/default.nix:22]: done too

**Comment by caarlos0** [comment]
hey! anything missing from me here?

**Pull request closed without merging**